enhancements/cnf-tests-split.md

```diff
@@ -0,0 +1,204 @@
+)
+```
+
+To prevent conflicts due to the numerous dependencies in the test suites, we want to decouple the external test suites by introducing a script that invokes the required test suites as git-submodules under the cnf-tests/submodules path.
+### Note
+1. In this proposal, we refer to each external test suite as a CNF component.
+2. Some dependencies will still be required in the components as there are tests that combine multiple features.
+
+
+This change will also allow us to pin the external repositories to the latest branch, ensuring that we always test the latest tests without the need for manual commit hash updates. We will include an option to set an environment variable that specifies the commit ref for each repo.
+
+For detailed information about the changes in the cnf-tests container, please refer to the section titled [Rewrite the cnf-tests container](#rewrite-the-cnf-tests-container).
+
+### Goals
+
+1. Stop vendoring the external test suites and run the tests directly from the projects.
+2. Maintain the same API for running the tests.
+3. Preserve the same testing report output.
+4. Refactor the cnf-tests container to contain only the latency-tests.
+
+### Non-Goals
+
+1. Rename existing tests to a certain standard.
+2. Change the scope of the tests being run.
+
+## Proposal
+### User Stories
+#### Story 1
+
+As a developer, I want to run the cnf-tests suites so that I can validate my CNF feature works properly.
+
+#### Story 2
+As an Openshift administrator, I want to run the latency tests so that
+I can benchmark different CNF features.
+
+### Workflow Description
+
+This change will not affect the current API, including parameters/environment variables used to invoke the cnf-tests: `TEST_SUITES`, `FEATURES`, `FOCUS_TESTS`, `SKIP_TESTS`, `GINKGO_PARAMS`, `FAIL_FAST`, and `TESTS_REPORTS_PATH`. 
+However, with this proposal, instead of running test binaries that combine all the CNF components,  we will introduce an **optional** parameter named `CNF_COMPONENTS` to control what external test suites to run.  
+By default, the test will run all of the components it ran prior to this change.
+
+To explain this new variable further, read the following example:
+Running the `e2esuite` test suite, with an empty `CNF_COMPONENTS` and `FEATURES=metallb`, will result in 
+running ginkgo's test in all of the components: sriov, metallb, nto-performance, nto-latency, ptp and integration, adding the flag `-ginkgo.focus=metallb`.
+So in other words, the functionallity of the `FEATURES` var stays the same.
+
+When running the `e2esuite` test suite with `CNF_COMPONENTS=metallb` it will invoke only the metallb component
+of the e2esuite, i.e.: `cnf-tests/submodules/metallb-operator/test/e2e/functional/tests`
+and for the validation suite, it will run: `cnf-tests/submodules/metallb-operator/test/e2e/validation/tests`
+
+The mapping of suite + component to the external test suite path is done by:
+```bash
+declare -A TESTS_PATHS=\
```

Comment:
Can we go with convention over configuration and expect the extenral suite to match the name of the local path of a given suite?

Reply:
Do you mean that the user will set EXTERNAL_SUITE to the local path e.g. `"cnf-tests/submodules/metallb-operator/test/e2e/validation"` instead of just metallb?
What are the benefits of going with the approach?

Reply:
I was thinking more something like "validation/metallb" (without the cnf-tests because it's implicit, without submodules because I am not sure it makes sense, see my comment below).
The advantage is not needing to maintain the map if we add / remove a suite.

Reply:
resolved the conversation in the comment below